### PR TITLE
CIS: Edge functions development

### PR DESCRIPTION
**Pull request by KumarGanesanIBM**

### CODE REVIEW

File: ibm/resource_ibm_cis_edge_functions_action.go [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+			},
+			cisDomainID: {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "CIS Domain ID",
+				DiffSuppressFunc: suppressDataDiff,
+			},
+			cisEdgeFunctionsActionScriptName: {
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (kavya498) [inline ibm/resource_ibm_cis_edge_functions_action.go:41]: Name cannot be updated.. Please add 
forceNew to the name

File: ibm/resource_ibm_cis_edge_functions_action.go [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+		return fmt.Errorf("Error: %v", err)
+	}
+	d.SetId(convertCisToTfThreeVar(scriptName, zoneID, crn))
+	return resourceIBMCISEdgeFunctionsActionRead(d, meta)
+}
+
+func resourceIBMCISEdgeFunctionsActionUpdate(d *schema.ResourceData, meta interface{}) error {
+	// No need to have upate function, since edge function script is going to be uploaded as a file
+	// so calling create method here.
+	return resourceIBMCISEdgeFunctionsActionCreate(d, meta)
```
Comment (kavya498) [inline ibm/resource_ibm_cis_edge_functions_action.go:89]: please return read here

File: ibm/resource_ibm_cis_edge_functions_action.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	if err != nil {
+		return fmt.Errorf("Error: %v", err)
+	}
+	d.SetId(convertCisToTfThreeVar(scriptName, zoneID, crn))
+	return resourceIBMCISEdgeFunctionsActionRead(d, meta)
+}
+
+func resourceIBMCISEdgeFunctionsActionUpdate(d *schema.ResourceData, meta interface{}) error {
+	// No need to have upate function, since edge function script is going to be uploaded as a file
+	// so calling create method here.
```
Comment (kavya498) [inline ibm/resource_ibm_cis_edge_functions_action.go:88]: please check for hasChange of `cisEdgeFunctionsActionScript ` and return create inside the condition block

File: ibm/resource_ibm_cis_edge_functions_action.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+		if err == io.EOF || n < 1 {
+			break
+		}
+	}
+	err = result.Close()
+	if err != nil {
+		return fmt.Errorf("Error in closing reader")
+	}
+
+	d.SetId(convertCisToTfThreeVar(scriptName, zoneID, crn))
```
Comment (kavya498) [inline ibm/resource_ibm_cis_edge_functions_action.go:123]: why is the id set again..?

File: ibm/resource_ibm_cis_edge_functions_action.go (hunk 3 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_edge_functions_action.go]: done

File: ibm/resource_ibm_cis_edge_functions_action.go (hunk 2 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_edge_functions_action.go]: done

File: ibm/resource_ibm_cis_edge_functions_action.go (hunk 1 above)
Comment (KumarGanesanIBM) [reply ibm/resource_ibm_cis_edge_functions_action.go:41]: done

**Pull request merged**